### PR TITLE
Prevent changing type in amendments with articulated or modifications

### PR DESCRIPTION
**Pull request by joseluispodemos**

### CODE REVIEW

File: app/models/amendments/articulated_amendment.rb [hunk 1]
```diff
@@ -5,7 +5,11 @@ class ArticulatedAmendment < Amendment
     has_one :articulated, foreign_key: 'amendment_id'
     delegate :section, to: :articulated, allow_nil: true
 
-    def allow_articulateds?
+    def articulated?
+      !articulated.nil?
```
Comment (leio10) [inline app/models/amendments/articulated_amendment.rb]: Try to not negate a negative question method. You can ask for `articulated` or, if you want to be more explicit, `articulated.present?`

File: app/models/amendment.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@
 class Amendment < ApplicationRecord
   belongs_to :user
   belongs_to :budget
+  validate :type_can_not_change, on: :update
```
Comment (leio10) [inline app/models/amendment.rb]: You can use a conditional validation here (and remove the call to `locked_type?` in line 29):

```ruby
validate :type_can_not_change, on: :update, if: :locked_type?
```
Also, you can call it `:type_cannot_be_changed` following the naming convention from [the guides](https://guides.rubyonrails.org/active_record_validations.html#custom-methods).

File: app/models/amendments/articulated_amendment.rb (hunk 1 above)
Comment (joseluispodemos) [reply app/models/amendments/articulated_amendment.rb]: Solved

File: app/models/amendment.rb (hunk 2 above)
Comment (joseluispodemos) [reply app/models/amendment.rb]: And solved too

**Pull request merged**